src/App.tsx

```diff
@@ -1,156 +1,240 @@
+import { ErrorNotification } from './components/ErrorMessage/ErrorNotification';
+
+const USER_ID = 7035;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoStatus, setTodoStatus] = useState(TodoStatus.ALL);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessage.NONE);
+  const [tempoTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [processingTodosIds, setProcessingTodosIds] = useState<number[]>([]);
+
+  const hasActiveTodo = todos.some((todo) => !todo.completed);
+  const inputsRef = useRef<HTMLInputElement>(null);
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (todoStatus) {
+        case TodoStatus.ACTIVE:
+          return !todo.completed;
+        case TodoStatus.COMPLETED:
+          return todo.completed;
+        default:
+          return true;
+      }
+    })
+    );
+  }, [todoStatus, todos]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const todosFromServer = await getTodos(USER_ID);
+
+      try {
+        setTodos(todosFromServer);
+      } catch {
+        setErrorMessage(ErrorMessage.LOAD);
+
+        setTimeout(() => {
+          setErrorMessage(ErrorMessage.NONE);
+        }, 2000);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const handleChangeStatus = useCallback(
+    async (todoID: number, data: Partial<Todo>) => {
+      try {
+        setProcessingTodosIds((prevIDs) => [...prevIDs, todoID]);
+
+        const updatedTodo = await patchTodo(todoID, data);
+
+        setTodos((prevTodos) => prevTodos.map((prevTodo) => {
+          if (prevTodo.id === todoID) {
+            return {
+              ...prevTodo,
+              ...updatedTodo,
+            };
+          }
+
+          return prevTodo;
+        }));
+      } catch {
+        setErrorMessage(ErrorMessage.UPDATE);
+      } finally {
+        setProcessingTodosIds([]);
+      }
+    }, [todos],
+  );
+
+  const handleChangeStatusAll = useCallback(async () => {
+    const allTodosPromises = todos.map((todo) => {
+      if ((hasActiveTodo && !todo.completed)
+        || (!hasActiveTodo && todo.completed)) {
+        setProcessingTodosIds((prev) => [...prev, todo.id]);
+      }
+
+      if (todos.every(task => task.completed)) {
+        return patchTodo(todo.id, { completed: false });
+      }
+
+      return patchTodo(todo.id, { completed: true });
+    });
+
+    try {
+      const updatedTodos = await Promise.all(allTodosPromises);
+
+      setTodos(updatedTodos);
+    } catch {
+      setErrorMessage(ErrorMessage.UPDATE);
+    } finally {
+      setProcessingTodosIds([]);
+    }
+  }, [todos]);
+
+  const createTodo = (title : string) => {
+    if (!title.trim().length) {
+      setErrorMessage(ErrorMessage.TITLE);
+    } else {
+      const NewTodo = {
```

Comment:
```suggestion
      const newTodo = {
```
camelCase for variable names